### PR TITLE
[Bugfix] Make preprocess compatible with openmpi

### PR DESCRIPTION
**Pull request by classicsong**
## Description
The current distributed partition preprocess only works with mpich. This PR extend it to support openmpi.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [x] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tools/distpartitioning/parmetis_preprocess.py [hunk 1]
```diff
@@ -37,8 +37,14 @@ def get_proc_info():
         Rank of the current process.
     """
     env_variables = dict(os.environ)
-    return int(os.environ.get("PMI_RANK") or 0)
-
+    # mpich
+    if "PMI_RANK" in env_variables:
```
Comment (Rhett-Ying) [inline tools/distpartitioning/parmetis_preprocess.py:41]: when `PMI_RANK` is introduced, I ask a question for it. https://github.com/dmlc/dgl/pull/4756/files#r1009055022 could you take a look? so it turns out that `PMI_RANK` is not universally suitable?

Comment (Rhett-Ying) [reply tools/distpartitioning/parmetis_preprocess.py:41]: could you pls update the docstring above as it's kind of out-dated?

Comment (classicsong) [reply tools/distpartitioning/parmetis_preprocess.py:41]: Open MPI use OMPI_COMM_WORLD_RANK.
It seems there is no uniform NAME.

Comment (Rhett-Ying) [reply tools/distpartitioning/parmetis_preprocess.py:41]: all right.

**Review by Rhett-Ying — APPROVED** [review]
pls update the out-dated part in the docstring of `get_proc_info()`

File: tools/distpartitioning/parmetis_preprocess.py (hunk 1 above)
Comment (classicsong) [reply tools/distpartitioning/parmetis_preprocess.py:32]: docstring updated.

**Pull request merged**